src_c/transform.c

```diff
@@ -521,66 +521,84 @@ stretch(SDL_Surface *src, SDL_Surface *dst)
+    SDL_Surface *retsurf = NULL;
 
     if (width < 0 || height < 0)
         return RAISE(PyExc_ValueError, "Cannot scale to negative size");
 
-    surf = pgSurface_AsSurface(surfobj);
+    src = pgSurface_AsSurface(srcobj);
 
-    if (!surfobj2) {
-        newsurf = newsurf_fromsurf(surf, width, height);
-        if (!newsurf)
+    if (!dstobj) {
+        retsurf = newsurf_fromsurf(src, width, height);
+        if (!retsurf)
             return NULL;
     }
-    else
-        newsurf = pgSurface_AsSurface(surfobj2);
+    else {
+        retsurf = pgSurface_AsSurface(dstobj);
+    }
 
-    if (newsurf->w != width || newsurf->h != height)
-        return RAISE(PyExc_ValueError,
-                     "Destination surface not the given width or height.");
+    if (retsurf->w != width || retsurf->h != height) {
+        return (SDL_Surface *)(RAISE(
+            PyExc_ValueError,
+            "Destination surface not the given width or height."));
+    }
 
-    /* check to see if the format of the surface is the same. */
-    if (surf->format->BytesPerPixel != newsurf->format->BytesPerPixel)
-        return RAISE(PyExc_ValueError,
-                     "Source and destination surfaces need the same format.");
+    if (src->format->BytesPerPixel != retsurf->format->BytesPerPixel) {
+        return (SDL_Surface *)(RAISE(
+            PyExc_ValueError,
+            "Source and destination surfaces need the same format."));
+    }
 
-    if ((width && height) && (surf->w && surf->h)) {
-        SDL_LockSurface(newsurf);
-        pgSurface_Lock(surfobj);
+    if ((width && height) && (src->w && src->h)) {
+        SDL_LockSurface(retsurf);
+        pgSurface_Lock(srcobj);
 
         Py_BEGIN_ALLOW_THREADS;
-        if (width == 2 * surf->w && height == 2 * surf->h) {
-            scale2xraw(surf, newsurf);
+        if (width == 2 * src->w && height == 2 * src->h) {
+            scale2xraw(src, retsurf);
         }
         else {
-            stretch(surf, newsurf);
+            stretch(src, retsurf);
         }
         Py_END_ALLOW_THREADS;
 
-        pgSurface_Unlock(surfobj);
-        SDL_UnlockSurface(newsurf);
+        pgSurface_Unlock(srcobj);
+        SDL_UnlockSurface(retsurf);
+    }
+
+    return retsurf;
+}
+
+static PyObject *
+surf_scale(PyObject *self, PyObject *args, PyObject *kwargs)
+{
+    pgSurfaceObject *surfobj;
+    pgSurfaceObject *surfobj2 = NULL;
+    PyObject *size;
+    SDL_Surface *newsurf;
+    int width, height;
+    static char *keywords[] = {"surface", "size", "dest_surface", NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "O!O|O!", keywords,
+                                     &pgSurface_Type, &surfobj, &size,
+                                     &pgSurface_Type, &surfobj2))
+        return NULL;
+
+    if (!pg_TwoIntsFromObj(size, &width, &height))
+        return RAISE(PyExc_TypeError, "size must be two numbers");
+
+    if (!(newsurf = scale_to(surfobj, surfobj2, width, height))) {
```

Comment:
I'd rather you assign and then check, rather than doing an assignment inside a comparison. It will compile to the same thing, but the former is more readable.